### PR TITLE
DXCDT-141: "Multi-environment Workflows" doc page [6/10]

### PR DESCRIPTION
**Pull request by willvedd**
## ✏️ Changes

The sixth of a ten PR cascade to publish the content from the Deploy CLI's documentation overhaul. This one focuses on the "Incorporating into multi-environment workflows" page.

## 🔗 References

- [Documentation content outline](https://oktawiki.atlassian.net/wiki/spaces/DXCDT/pages/2632975220/Deploy+CLI+Docs+Content+Outline)
- [I"ncorporating into Mult-environment Workflows" draft](https://oktawiki.atlassian.net/wiki/spaces/DXCDT/pages/2650711997/Incorporating+into+multi-environment+workflows)

### CODE REVIEW

File: docs/multi-environment-workflow.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Incorporating into multi-environment workflows
+
+The Deploy CLI is most powerful when working within a multi-tenant, multi-environment context. When integrated into your CI/CD development workflows, can be used to propagate Auth0 changes from feature development all the way through production.
```
Comment (sergiughf) [inline docs/multi-environment-workflow.md:3]:
```suggestion
The Deploy CLI supports working within a multi-tenant, multi-environment context. When integrated into your CI/CD development workflows, can be used to propagate Auth0 changes from feature development all the way through production.
```

IMO the wording implies that it shouldn't be used for anything else than multi env context.

File: docs/multi-environment-workflow.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+| Development | travel0-dev   |
+| Testing     | travel0-uat   |
+| Staging     | travel0-stage |
+| Production  | travel0-prod  |
+
+## Resource configuration repository
+
+When exported, your Auth0 tenant state will be represented as a set of resource configuration files, either in a [YAML or JSON format](#). In a multi-environment context it is expected to have a single repository of resource configurations that is applied to all environments. In practice, this may exist as a directory in your project’s codebase or in a separate codebase altogether.
+
+You should have at least one branch for each tenant in your repository, which allows you to make changes without deploying them (the changes would only deploy when you merged your branch into the master, or primary, branch). With this setup, you can have a continuous integration task for each environment that automatically deploys changes to the targeted environment whenever the master branch receives updates.
```
Comment (sergiughf) [inline docs/multi-environment-workflow.md:26]: I find this a bit confusing personally. Couldn't I have a branch that updates all the env configs and have both of them deploy when it gets merged to master?

File: docs/multi-environment-workflow.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+Your workflow would, therefore, look something like this:
```
Comment (sergiughf) [inline docs/multi-environment-workflow.md:28]:
```suggestion
Your workflow could potentially look something like this:
```

Suggestion

File: docs/multi-environment-workflow.md (hunk 2 above)
Comment (willvedd) [reply docs/multi-environment-workflow.md:26]: >Couldn't I have a branch that updates all the env configs and have both of them deploy when it gets merged to master?

That would defeat the purpose of the multi-env workflow. By incrementally propagating changes from lower environments all the way to production, you're able to test that your changes work. If you were to change _all_ environment configurations when merging to main, you're deploying to dev, staging and prod all at once.

I think some confusion here might stem from "env config" being named similarly to "resource config".

File: docs/multi-environment-workflow.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
+# Incorporating into multi-environment workflows
+
+The Deploy CLI supports working within a multi-tenant, multi-environment context. When integrated into your CI/CD development workflows, can be used to propagate Auth0 changes from feature development all the way through production.
+
+In general, the advised workflow is as follows:
+
+- Create a separate Auth0 tenant for each environment (ex: dev, staging, prod)
```
Comment (Widcket) [inline docs/multi-environment-workflow.md:7]: Maybe these could be an ordered list?

File: docs/multi-environment-workflow.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
+# Incorporating into multi-environment workflows
```
Comment (Widcket) [inline docs/multi-environment-workflow.md:1]: I'd suggest adding a table of contents

**Pull request merged**